packages/salesforcedx-vscode-replay-debugger/src/states/variableAssignmentState.ts

```diff
@@ -26,106 +26,128 @@ export class VariableAssignmentState implements DebugLogState {
       const nameSplit = this.fields[3].split('.');
       const className =
-        name.indexOf('.') > -1
-          ? name.substring(0, name.lastIndexOf('.'))
-          : name;
+        name.indexOf('.') > -1 ? name.substring(0, name.lastIndexOf('.')) : '';
       const varName =
         nameSplit.length > 0 ? nameSplit[nameSplit.length - 1] : name;
       const value = this.fields[4];
-      let ref = '0';
+      let ref;
       if (this.fields.length === 6) {
         ref = this.fields[5];
       }
+
+      const refMap = logContext.getRefsMap();
       let container: ApexVariableContainer | undefined;
       let map: Map<String, VariableContainer> | undefined;
+      let isNested = false;
+
       if (logContext.getStaticVariablesClassMap().has(className)) {
         map = logContext.getStaticVariablesClassMap().get(className)!;
         container = map.get(varName)! as ApexVariableContainer;
       } else if (frameInfo.locals.has(varName)) {
         // if name does not contain '.' (i.e. this.attr or a.Name), it should be in locals and we can update the value
         map = frameInfo.locals;
         container = map.get(varName) as ApexVariableContainer;
+      } else if (name.indexOf('.') !== -1) {
+        isNested = true;
       }
 
-      if (
-        container &&
-        map &&
-        container.variablesRef !== 0 &&
-        !logContext.getRefsMap().has(ref)
-      ) {
-        container = new ApexVariableContainer(
-          container.name,
-          '',
-          container.type
-        );
-        map.set(varName, container);
-      }
-
-      if (container) {
-        container.value = value;
-      }
-
-      // assigning to top level variable in locals
-      if (
-        ref !== '0' &&
-        container &&
-        !container.type.startsWith('Map<') &&
-        !container.type.startsWith('List<') &&
-        !container.type.startsWith('Set<')
-      ) {
-        if (!logContext.getRefsMap().has(ref)) {
-          logContext.getRefsMap().set(ref, container);
-        } else {
-          // if the ref already exists then that means we are assigning another pointer to a ref so we need to merge the info from the ref into the variable container
-          const refContainer = logContext.getRefsMap().get(ref)!;
-          container.variables = refContainer.variables;
-          container.variablesRef = refContainer.variablesRef;
-          container.name = varName;
+      if (ref) {
+        // update the refcontainer mapping
+        if (!refMap.has(ref)) {
+          logContext
+            .getRefsMap()
+            .set(ref, new ApexVariableContainer('', '', '', ref));
         }
-        if (value.indexOf('{') === 0 && value !== '{}') {
-          container.value = '';
-          container.variablesRef = logContext
-            .getVariableHandler()
-            .create(container);
-          this.parseAndPopulate(value, container);
+        const refContainer = refMap.get(ref)!;
+        if (value !== '{}') {
```

Comment:
I refactored this a bit to arrange it in a way that made sense. Previously there were too many conditions to keep track of and it wasn't straight forward when you had a variable container and how you got it. I hope this is a little easier to follow

Reply:
@Sweetman , would it help if you add some comments in this area so the next time you revisit, you're recognizing the scenarios instead of recalling?

Reply:
Added